### PR TITLE
rust-analyzer for non-Cargo projects

### PR DESCRIPTION
**Pull request by skyfmmf**
rust-analyzer can also be used in non-cargo projects. This requires a rust-project.json file in the project root (https://rust-analyzer.github.io/manual.html#non-cargo-based-projects).

Make the rust-analyzer linter search for a rust-project.json file if no Cargo.toml file could be found; document it; and add tests for it.

### CODE REVIEW

File: ale_linters/rust/analyzer.vim [hunk 1]
```diff
@@ -9,9 +9,19 @@ function! ale_linters#rust#analyzer#GetCommand(buffer) abort
 endfunction
 
 function! ale_linters#rust#analyzer#GetProjectRoot(buffer) abort
+    " Try to find nearest Cargo.toml for cargo projects
     let l:cargo_file = ale#path#FindNearestFile(a:buffer, 'Cargo.toml')
+    if !empty(l:cargo_file)
```
Comment (w0rp) [inline ale_linters/rust/analyzer.vim:14]:
```suggestion

    if !empty(l:cargo_file)
```

File: ale_linters/rust/analyzer.vim [hunk 2]
```diff
@@ -9,9 +9,19 @@ function! ale_linters#rust#analyzer#GetCommand(buffer) abort
...
+        return fnamemodify(l:cargo_file, ':h')
+    endif
 
-    return !empty(l:cargo_file) ? fnamemodify(l:cargo_file, ':h') : ''
+    " Try to find nearest rust-project.json for non-cargo projects
+    let l:rust_project = ale#path#FindNearestFile(a:buffer, 'rust-project.json')
+    if !empty(l:rust_project)
```
Comment (w0rp) [inline ale_linters/rust/analyzer.vim:20]:
```suggestion

    if !empty(l:rust_project)
```

**Review by w0rp — CHANGES_REQUESTED** [review]
I left some comments on syntax where we require blank lines between blocks and statements. Have a look at the failing tests.

**Review by hsanson — APPROVED** [review]
Looks good, thanks.

**Pull request merged**